File: backend/backend-springboot/src/main/java/ec/edu/espe/bancaMovil/controller/CardController.java
package ec.edu.espe.bancaMovil.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import ec.edu.espe.bancaMovil.model.Card;
import ec.edu.espe.bancaMovil.service.CardService;
import java.util.List;
@RestController
@RequestMapping("/api/cards")
public class CardController {
    @Autowired
    private CardService cardService;

    // Agregar una tarjeta
    @PostMapping("/")
    public Card addCard(@RequestBody Card card) {
        return cardService.addCard(card);
    }

    // Congelar/Descongelar una tarjeta
    @PutMapping("/{cardId}/freeze")
    public void freezeCard(@PathVariable Long cardId, @RequestBody boolean isFrozen) {
        cardService.freezeCard(cardId, isFrozen);
    }

    // Eliminar una tarjeta
    @DeleteMapping("/{cardId}")
    public void deleteCard(@PathVariable Long cardId) {
        cardService.deleteCard(cardId);
    }

    // Obtener todas las tarjetas de un usuario
    @GetMapping("/user/{userId}")
    public List<Card> getCardsByUser(@PathVariable Long userId) {
        return cardService.getCardsByUser(userId);
    }

    // Obtener una tarjeta por Id
    @GetMapping("/{cardId}")
    public Card getCardById(@PathVariable Long id) {
        return cardService.getCardById(id);
    }
}

File: frontend/lib/controller/card_controller.dart
import 'package:flutter/material.dart';
import '../services/card_service.dart';
import '../models/card_model.dart';

class CardController extends ChangeNotifier {
  final CardService _service = CardService();
  List<CardModel> _cards = [];

  List<CardModel> get cards => _cards;

  Future<void> fetchCards(String userId) async {
    _cards = await _service.getCardsByUser(userId);
    notifyListeners();
  }

  Future<void> addCard(CardModel card) async {
    CardModel newCard = await _service.addCard(card);
    _cards.add(newCard);
    notifyListeners();
  }

  Future<void> freezeCard(String cardId, bool isFrozen) async {
    await _service.freezeCard(cardId, isFrozen);
    int index = _cards.indexWhere((c) => c.id == cardId);
    if (index != -1) {
      _cards[index] = _cards[index].copyWith(isFrozen: isFrozen);
      notifyListeners();
    }
  }

  Future<void> removeCard(String cardId) async {
    await _service.deleteCard(cardId);
    _cards.removeWhere((c) => c.id == cardId);
    notifyListeners();
  }

  Future<CardModel?> getCardById(String cardId) async {
    return await _service.getCardById(cardId);
  }
}
